packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -240,21 +312,108 @@ export class Builder extends EventEmitter {
     }
   }
 
-  private async updateDocumentForFile(file: string) {
-    const contents = await readFile(file, 'utf8');
-    if (contents.trim().length === 0) {
-      return;
+  private async updateDocumentForFile(filePath: string) {
+    const project = this.config.getConfigForFile(filePath);
+
+    if (project) {
+      const contents = await readFile(filePath, 'utf8');
+      if (contents.trim().length === 0) {
+        return undefined;
+      }
+
+      const document = parse(new Source(contents, filePath));
+      this.documentCache.set(filePath, document);
+
+      return document;
+    }
+
+    return undefined;
+  }
+
+  private getProjects() {
+    const projects = this.config.getProjects();
+
+    if (projects) {
+      return Object.values(projects);
     }
 
-    const document = parse(new Source(contents, file));
-    this.documentCache.set(file, document);
+    this.emit('error', new Error(`No projects configured in .graphqlconfig`));
+
+    return [];
+  }
+
+  private getGlobs() {
+    return this.getProjects().reduce<string[]>((glob, project) => {
+      return glob.concat(project.includes);
+    }, []);
+  }
+
+  private getSchemaPaths() {
+    return this.getProjects().reduce<string[]>((schemas, x) => {
+      if (x.schemaPath) {
+        schemas.push(x.schemaPath);
+      }
+      return schemas;
+    }, []);
+  }
+
+  private getProjectDocuments(): GraphQLProjectDocuments {
+    return Array.from(this.documentCache.entries())
+      .map(([filePath, document]) => ({
+        document,
+        filePath,
+        project: this.config.getConfigForFile(filePath),
+      }))
+      .reduce((groups, {document, filePath, project}) => {
+        if (project) {
+          const group = groups.get(project.projectName || '');
+
+          if (group) {
+            group.set(filePath, document);
+          } else {
+            groups.set(
+              project.projectName || '',
+              new Map<string, DocumentNode>([[filePath, document]]),
+            );
+          }
+        }
+        return groups;
+      }, new Map<string, Map<string, DocumentNode>>());
   }
 
   private removeDocumentForFile(file: string) {
     this.documentCache.delete(file);
   }
 }
 
+function getProjectForSchemaPath(config: GraphQLConfig, schemaPath: string) {
+  const project = Object.values(config.getProjects() || {})
+    .filter((project) => project.schemaPath === schemaPath)
+    .shift();
+
+  if (project) {
+    return project;
+  }
+
+  throw new Error(
+    `No project defined in .graphqlconfig for schema ${schemaPath}`,
+  );
+}
+
+function getSchemaTypesPath(project: GraphQLProjectConfig) {
+  if (typeof project.extensions.schemaTypesPath === 'string') {
+    return project.extensions.schemaTypesPath;
+  }
+
+  if (project.schemaPath) {
+    return project.schemaPath.replace(/\.json$/, '.types.ts');
```

Comment:
this was just a sane default choice for how to format schema types, open to suggestions if we want to change it.

Reply:
I just noticed that this fails if you provide a `*.graphql` schema path